### PR TITLE
Re-design the parameter transfer logic between thread ctx switch.

### PR DESCRIPTION
**Pull request by xyzhao2018**
Signed-off-by: Shirley Zhao <shirley.zhao@intel.com>

### CODE REVIEW

File: core/arch/x86_64/kernel/thread_64.S [hunk 1]
```diff
@@ -205,30 +243,61 @@ FUNC thread_rpc , :
...
+    pushq %rdi
+
     movq $OPTEE_SMC_RETURN_RPC_FOREIGN_INTR, %r10
 
     movq %rdi, %r11
 
     movq $0x0, %r12
 
-    pushfq
+    #pushfq
```
Comment (GangSecurity) [inline core/arch/x86_64/kernel/thread_64.S]: remove the line

File: core/arch/x86_64/kernel/thread_64.S [hunk 2]
```diff
@@ -205,30 +243,61 @@ FUNC thread_rpc , :
...
 
-    pushq %rbp
+    #pushq %rbp
```
Comment (GangSecurity) [inline core/arch/x86_64/kernel/thread_64.S]: remove the line

File: core/arch/x86_64/kernel/thread_64.S (hunk 1 above)
Comment (xyzhao2018) [reply core/arch/x86_64/kernel/thread_64.S]: Done

File: core/arch/x86_64/kernel/thread_64.S (hunk 2 above)
Comment (xyzhao2018) [reply core/arch/x86_64/kernel/thread_64.S]: Done

**Review by jingdlu — APPROVED** [review]

**Review by GangSecurity — APPROVED** [review]

**Pull request merged**